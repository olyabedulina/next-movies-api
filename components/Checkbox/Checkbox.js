import React from 'react'
import PropTypes from 'prop-types'

import CM from './styles.module.css'

const Checkbox = ({
    className = '',
    checked = false,
    onChange = Function.prototype,
    data,
    children
}) => {

    function handleChange(event) {
         onChange(!checked, data)
    }

    return <label className={CM.checkboxLabel}>
            <input
                name={name}
                className={`${className} ${CM.checkbox}`}
                type="checkbox"
                checked={checked}
                onChange={handleChange}
            />
            <span className={CM.checkboxText}>{children}</span>
        </label>
}

Checkbox.propTypes = {
    className: PropTypes.string,
    checked: PropTypes.bool,
    onChange: PropTypes.func,
    data: PropTypes.string,
    children: PropTypes.string
};


export default Checkbox
