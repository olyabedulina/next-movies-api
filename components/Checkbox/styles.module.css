.checkbox {
  position: absolute;
  z-index: -1;
  opacity: 0;
}

.checkboxText {
  color: #fff;
  font-size: 18px;
  line-height: 20px;
  cursor: pointer;
}

.checkbox + .checkboxText {
  display: inline-flex;
  align-items: center;
  user-select: none;
}

.checkbox + .checkboxText:before {
  content: '';
  display: inline-block;
  width: 0.9em;
  height: 0.9em;
  flex-shrink: 0;
  flex-grow: 0;
  border: 1px solid #6c6d6f;
  border-radius: 0.25em;
  margin-right: 6px;
  background-color: #fff;
  background-repeat: no-repeat;
  background-position: center center;
  background-size: 50% 50%;
}

.checkbox:checked + .checkboxText:before {
  border-color: #e64c5a;
  box-shadow: 0 0 2px 0 #1b1b1b;
  background-color: #e64c5a;
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 8 8'%3e%3cpath fill='%23fff' d='M6.564.75l-3.59 3.612-1.538-1.55L0 4.26 2.974 7.25 8 2.193z'/%3e%3c/svg%3e");
}

.checkbox:not(:disabled):not(:checked) + .checkboxText:hover::before {
  border-color: #b3d7ff;
}

.checkbox:not(:disabled):not(:checked):active + .checkboxText:before {
  border-color: #b3d7ff;
}

.checkbox:focus + .checkboxText:before {
  box-shadow: 0 0 2px 0 rgba(0, 123, 255, 0.25);
}

.checkbox:focus:not(:checked) + .checkboxText:before {
  border-color: #80bdff;
}

.checkbox:disabled + .checkboxText:before {
  background-color: #e9ecef;
}