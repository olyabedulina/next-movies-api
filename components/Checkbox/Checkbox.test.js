import React from 'react';
import renderer from 'react-test-renderer';

import Checkbox from "./Checkbox";

describe('Components :: Checkbox snapshots:', () => {
    it('Test :: Checkbox component', () => {
        const actual = renderer
            .create(<Checkbox checked={true} data='1'>Test name</Checkbox>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Test :: Checkbox component', () => {
        const actual = renderer
            .create(<Checkbox checked={false} data='Abc'>Checkbox</Checkbox>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Test :: Checkbox component', () => {
        const actual = renderer
            .create(<Checkbox />)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });
});
