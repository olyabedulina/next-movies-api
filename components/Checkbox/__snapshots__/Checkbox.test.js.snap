// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components :: Checkbox snapshots: Test :: Checkbox component 1`] = `
<label>
  <input
    checked={true}
    className=" checkbox"
    name=""
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="checkboxText"
  >
    Test name
  </span>
</label>
`;

exports[`Components :: Checkbox snapshots: Test :: Checkbox component 2`] = `
<label>
  <input
    checked={false}
    className=" checkbox"
    name=""
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="checkboxText"
  >
    Checkbox
  </span>
</label>
`;

exports[`Components :: Checkbox snapshots: Test :: Checkbox component 3`] = `
<label>
  <input
    checked={false}
    className=" checkbox"
    name=""
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="checkboxText"
  />
</label>
`;
